kitsune/wiki/tests/test_views.py

```diff
@@ -684,3 +685,170 @@ def test_it_works(self):
+        """
+        Test that the article is showing in the header locale
+
+        Test while the document is not translated to requested locale but translated to
+        one of the locale mentioned in ACCEPT_LANGUAGE Header
+        """
+        # A Client with es, fr, ja language choices in ACCEPT_LANGUAGE Header
+        header = 'es,fr;q=0.7,ja;q=0.3,'
+        # Create a Translated document in fr locale version.
+        # And try to get data of es locale vesion of document with above mentioned client
+        doc_content = self.get_data_from_translated_document(header=header,
+                                                             create_doc_locale='fr',
+                                                             req_doc_locale='es')
+        # As the document is not translated into es locale version, but in the client header
+        # there is fr locale and the document is translated into fr locale, the user will see
+        # fr version of the document.
+        en_content = 'This article is in English'
+        trans_content = 'This article is translated into fr'
+        assert en_content not in doc_content
+        assert trans_content in doc_content
+
+    def test_header_locales_not_in_translated_list(self):
+        """
+        Test the article is showing in the fallback locale of header locele
+
+        * The article is not translated into requested locale
+        * The article is not translated to any of the locale of ACCEPT_LANGUAGE header
+        * But we have a fallback locale for one of the locales of ACCEPT_LANGUAGE header
+           ** This fallback locale is mentioned in kitsune/settings.py
+        *** Test that the fallback locale is being used
+        """
+
+        # A Client with de, an, ja language choices in ACCEPT_LANGUAGE Header
+        header = 'de,an;q=0.7,ja;q=0.3,'
+        # Create a translated document in es locale version.
+        # And try to get data of de locale vesion of document with above mentioned client
+        doc_content = self.get_data_from_translated_document(header=header,
+                                                             create_doc_locale='es',
+                                                             req_doc_locale='de')
+        # As the document is not translated into de locale version, also there are no locale
+        # in client header which the document is translated. But we have a fallback locale
+        # mentioned in kitsune/settings.py for the an locale. it is es. As the document is
+        # translated into es version, the user should see es version of the document
+        en_content = 'This article is in English'
+        trans_content = 'This article is translated into es'
+        assert en_content not in doc_content
+        assert trans_content in doc_content
+
+    def test_custom_locale_mapping(self):
+        """
+        Test that configured locale fallbacks are used for requested locale
```

Comment:
Entire section rewrite:

"Test that the article is showing according to defined locale fallback options (more than one), with the following conditions:

    the article is not localized into the user-requested locale
    the article is not localized into any of the locales listed in the ACCEPT_LANGUAGE header
    none of the ACCEPT_LANGUAGE header locales has a defined fallback locale
    the user-requested locale has more than one fallback locale defined and the requested article is localized into one of the defined fallback locales
    the fallback locale is set in kitsune/settings.py

Test that the article is shown in the defined fallback locale"